legend-engine-xts-mastery/legend-engine-xt-mastery-grammar/src/main/java/org/finos/legend/engine/language/pure/dsl/mastery/grammar/from/MasteryParseTreeWalker.java

```diff
@@ -841,4 +869,18 @@ private String visitQualifiedName(MasteryParserGrammar.QualifiedNameContext qual
     {
         return PureGrammarParserUtility.fromQualifiedName(qualifiedNameContext.packagePath() == null ? Collections.emptyList() : qualifiedNameContext.packagePath().identifier(), qualifiedNameContext.identifier());
     }
+
+    private void validateMasteryRuntime(List<PackageableElement> packageableElements)
+    {
+        List<MasteryRuntime> masteryRuntimes = ListIterate.selectInstancesOf(packageableElements, MasteryRuntime.class);
+        //validate master record definitions is unique across all mastery runtimes
+        Set<String> uniqueMasterRecordDefinitions = new HashSet<>();
+        List<String> masterRecordDefinitions = ListIterate.flatCollect(masteryRuntimes, m -> m.masterRecordDefinitions);
+        List<String> duplicateMasterRecords = ListIterate.select(masterRecordDefinitions, m -> !uniqueMasterRecordDefinitions.add(m));
+        if (!duplicateMasterRecords.isEmpty())
+        {
+            throw new EngineException(format("The following master record definitions %s appear multiple times. There cannot be duplicate masterRecordDefinitions across one or multiple mastery runtimes.", duplicateMasterRecords), EngineErrorType.PARSER);
```

Comment:
Can we make the error message consistent? master record definition or masterRecordDefinition?